src/partial-templates/partial-templates.service.ts

```diff
@@ -85,13 +103,72 @@ export class PartialTemplatesService {
    * @returns {Promise<PartialTemplate>} - The deleted partialTemplate
    */
   async remove(id: number): Promise<{ message: string }> {
-    // Ensures the partial template exists before deleting
     await this.findOne(id);
     await this.prisma.partialTemplate.delete({
       where: {
         id
       }
     });
-    return { message: 'Deleted successfully' };
+    return { message: 'Eliminado con exito' };
   }
+
+   /**
+   * Validates if partailTemplate ID exists
+   * @param {number} id - partailTemplate ID to validate
+   * @returns {Promise<Template>} - The partailTemplate with the given ID
+   * @throws {NotFoundException} - If the partailTemplate ID is not found
+   */
+  async validateId(id: number): Promise<PartialTemplate> {
+    const partialTemplate = await this.prisma.partialTemplate.findUnique({
+      where: {
+        id,
+      },
+    });
+
+    if (!partialTemplate) {
+      throw new NotFoundException('Plantilla Parcial no encontrada');
+    }
+    return partialTemplate;
+  }
+
+  /**
+   * Validates if foreign keys (templateId)
+   * @param {CreatePartialTemplateDto | UpdatePartialTemplateDto} dto - Data to validate
+   * @throws {NotFoundException} - If any foreign keys is not found
+   */
+  private async validateForeignKeys( dto: CreatePartialTemplateDto | UpdatePartialTemplateDto): Promise<void> {
+    const { templateId } = dto;
+    const validations = [];
+
+    if (templateId !== undefined) {
+      validations.push(
+        this.prisma.area.count({ where: { id: templateId } }).then((count) => {
+          if (count === 0) {
+            throw new NotFoundException(`Plantilla con id ${templateId} no encontrada`);
+          }
+        }),
+      );
+    }
+    await Promise.all(validations);
+  }
+
+  /**
+   * Restrict total hours by position 
+   * @param {CreatePartialTemplateDto | UpdatePartialTemplateDto} dto - Data to validate
+   * @throws {BadRequestException} - If the total do not match
+   */
+  private validateTotalByPosition(dto: CreatePartialTemplateDto | UpdatePartialTemplateDto): void {
+    const { position, total } = dto;
+
+    if(position === 'Profesor de asignatura'){
+      if(total < 17 || total > 32 ){
+        throw new BadRequestException('Cantidad de horas no permitida');
+      }
+    } else if (position === 'Profesor de Tiempo Completo'){
```

Comment:
I suggest to use
```ts
position.toLowerCase().includes('tiempo completo')
position.toLowerCase().includes('de asignatura')
```
just to cover most of the cases where is "Profesora de asignatura" or "Profesor Asociado de Tiempo Completo"